src/browser/BrowserService.cpp

```diff
@@ -546,6 +513,80 @@ void BrowserService::removeStoredPermissions()
     }
 }
 
+void BrowserService::convertAttributesToCustomData()
+{
+    if (!isDatabaseOpened()) {
+        QMessageBox::critical(0, tr("KeePassXC: Database locked!"),
+                              tr("The active database is locked!\n"
+                                 "Please unlock the selected database or choose another one which is unlocked."),
+                              QMessageBox::Ok);
+        return;
+    }
+
+    Database* db = m_dbTabWidget->currentDatabaseWidget()->database();
```

Comment:
Why isn't "getDatabase()" used here?

Here is my problem with this settings dialog, application settings are meant to be global. So these buttons make no sense at all. Basically I have to have the correct database selected prior to hitting the settings button, which is an undiscoverable "feature".

Here are some options:
1) Move everything below "Sort matching credentials..." into the database settings
2) Add a drop-down selection list above the buttons that let's you select from the currently open databases.

To me, option 1 is the most acceptable because it makes sense. It is also where the KeeShare settings are going so that would be consistent.

Reply:
A separate Browser Integration page should be added to the Database settings. Then it would be nice and clean.